parts/linux/cloud-init/artifacts/ipv6_nftables.service

```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=Configure nftables rules for handling Azure SLB IPv6 health probe packets
+
+[Service]
```

Comment:
this allows service invocation and restart on failure, but is of Type=simple and so will default to assuming execution succeeds as soon as `fork()` returns, meaning we won't actually track success of the script

suggest the workaround in https://unix.stackexchange.com/questions/272313/how-to-assure-a-oneshot-service-gets-retried-if-it-fails-the-first-time, making the service `Type=notify`, with `systemd-notify --ready` after the script succeeds? I don't see a better way for the same restart behavior with simple/oneshot. Exec gets you the error handling, but systemd won't wait for it to exit like oneshot.

Reply:
this is also relatively minor, I mean if the script fails the traffic will just break regardless of service type, this is mostly to provide a good signal (systemd unit failing) that something is wrong.

Reply:
latest update is `Type=notify` with the `systemd-notify --ready` at the end of the script. Tested on node and seems to work fine.

So is the tl;dr that this makes it easier to troubleshoot if this service is failing because it will stop other services from starting before it is finished?

Reply:
Sorry, brain shortcircuited. I don't think you actually need Type=notify, although it won't hurt. 

> is the tl;dr that this makes it easier to troubleshoot if this service is failing because it will stop other services from starting before it is finished

right. but, there are no dependent services :D

My thinking for context:
- Type=simple considered successfully executed after fork()
- Type=exec ... after exec()
- Type=oneshot ... after exit with non-zero code
- Type=notify...after notification sent for readiness

The think my brain lost track of, is of course systemd will consider the unit failed if it exits non-zero regardless. Since there are no deps, there's not much point blocking initial startup, you can just let it fail execution and enter failed state regardless of service type (but probably still want something other than oneshot for Restart=on-failure on e.g. imds failures or something).

Reply:
makes sense, I'll move it back to `simple` with the restarts.